src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -193,14 +200,15 @@ private IISAwaitable DoFlushAsync()
             unsafe
             {
                 var hr = 0;
-                hr = NativeMethods.http_flush_response_bytes(_pHttpContext, IISAwaitable.FlushCallback, (IntPtr)_thisHandle, out var fCompletionExpected);
-
+                hr = NativeMethods.http_flush_response_bytes(_pHttpContext, out var fCompletionExpected);
                 if (!fCompletionExpected)
                 {
-                    CompleteFlush(hr, 0);
+                    FreePinnedHeaders(_pinnedHeaders);
```

Comment:
Does this get freed in 2 places?

Reply:
Nope. This free happens on synchronous completion and the other happens on async completion.